addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -732,6 +732,20 @@ export class OdooEditor extends EventTarget {
         step.mutations = step.mutations.slice(0, until);
         this._toRollback = false;
     }
+    /**
+     * Undo the current non-recorded draft step.
+     */
+    historyRevertDraft() {
+        const draftStep = this._currentStep;
+        this.observerFlush();
+        this.historyRevert(draftStep, {sideEffect: false});
+        this.observerFlush();
+        // Clear draft step from all previous changes.
+        draftStep.mutations = [];
+
+        this._activateContenteditable();
+        this.historySetSelection(draftStep);
+    }
```

Comment:
I'll leave this for editor approval later too, I am not sure to understand the bug.

Reply:
This was extracted to #83114 and has been merged.